pilot/pkg/xds/delta.go

```diff
@@ -561,6 +564,13 @@ func requiresResourceNamesModification(url string) bool {
 	return url == v3.AddressType || url == v3.WorkloadType
 }
 
+// neverRemoveDelta checks if a type should never remove resources
+func neverRemoveDelta(url string) bool {
+	// https://github.com/envoyproxy/envoy/issues/32823
+	// We want to garbage collect extensions when they are no longer referenced, rather than delete immediately
+	return url == v3.ExtensionConfigurationType
```

Comment:
That issue mentions other top-level types; should we never remove those either?

Reply:
ECDS is not a top level resource, only CDS and LDS are. The rest of the dependant resources (that we use in Istio at least) do not handle deletes like ECDS. Instead delete basically means "You requested this and we don't have it".

Maybe we still need a way to express that for ECDS though...

Reply:
Not sure I understand this formulation; isn't the crux of the issue that ECDS deletes are applied while listener is draining?

Reply:
So consider a route, which is also tied to a listener. If you send "Remove route X" that is still referenced by a (presumably, draining) listener, it is entirely ignored (_any_ "Remove" is ignored, its not just about draining).
SDS works the same


ECDS is different, and instantly deletes it causing 500 errors (NFCF) for requests during the drain

Reply:
Ack - in that case I think this is fine; we'll just assume that the removal is surfaced via the listener removing it